### PR TITLE
refactor: generalize & slim down role permissions

### PR DESCRIPTION
**Pull request by ericpaulsen**
this PR builds on the work done in #252 and the feedback in #269. I'm leaving this in a draft state as we work to identify & remove unnecessary permissions.

ideally this should slim down the role to only the bare minimum permissions, as to enable smoother installs on OpenShift and any K8s distro.

### CODE REVIEW

File: templates/rbac.yaml [hunk 1]
```diff
@@ -50,12 +50,24 @@ metadata:
     app.kubernetes.io/version: {{ .Chart.AppVersion }}
     app.kubernetes.io/component: {{ include "coder.serviceName" . }}
 rules:
-  - apiGroups: ["", "apps", "networking.k8s.io"] # "" indicates the core API group
-    resources: ["persistentvolumeclaims", "pods", "deployments", "services", "secrets", "pods/exec","pods/log", "events", "networkpolicies", "serviceaccounts"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims", "pods", "services", "secrets", "serviceaccounts", "pods/exec"]
```
Comment (spikecurtis) [inline templates/rbac.yaml:54]: I highly doubt all these verbs are sensible on "pods/exec"

**Comment by ericpaulsen** [comment]
@spikecurtis I've deployed a test Coder instance using the modified `rbac.yaml` file in this PR. The basic functionality works as expected. Specifically, workspace builds with:

- CVMs on/off
- Self-contained builds on/off
- Custom StorageClasses

**Comment by johnstcn** [comment]
> @spikecurtis I've deployed a test Coder instance using the modified `rbac.yaml` file in this PR. The basic functionality works as expected. Specifically, workspace builds with:
> 
> * CVMs on/off
> * Self-contained builds on/off
> * Custom StorageClasses

I'm standing up a local OpenShift box to test against

**Comment by johnstcn** [comment]
This worked for me on OKD 4.11 (using CodeReady Containers) after installing the metrics-server Helm chart from https://kubernetes-sigs.github.io/metrics-server/

**Review by johnstcn — APPROVED** [review]

File: templates/rbac.yaml [hunk 2]
```diff
@@ -50,12 +50,26 @@ metadata:
...
     verbs: ["create", "get", "list", "watch", "update", "patch", "delete", "deletecollection"]
-  - apiGroups: ["metrics.k8s.io", "storage.k8s.io"]
-    resources: ["pods", "storageclasses"]
+  - apiGroups: [""]
+    resources: ["pods/log", "events"]
     verbs: ["get", "list", "watch"]
+  - apiGroups: ["apps"]
+    resources: ["deployments"]
+    verbs: ["create", "get", "list", "watch", "update", "patch", "delete", "deletecollection"]
+{{- if .Capabilities.APIVersions.Has "metrics.k8s.io/v1beta1" }}
```
Comment (johnstcn) [inline templates/rbac.yaml:62]: review: https://helm.sh/docs/chart_template_guide/builtin_objects/#:~:text=Capabilities%3A%20This%20provides%20information%20about%20what%20capabilities%20the%20Kubernetes%20cluster%20supports.

File: templates/rbac.yaml (hunk 1 above)
Comment (johnstcn) [reply templates/rbac.yaml]: I broke this out into a separate rule. I think the only verb possible here is `create pods/exec`, correct?

**Review by spikecurtis — APPROVED** [review]

**Pull request merged**